moveit_core/planning_scene/src/planning_scene.cpp

```diff
@@ -1062,24 +1062,29 @@ void planning_scene::PlanningScene::loadGeometryFromStream(std::istream &in, con
 
 void planning_scene::PlanningScene::setCurrentState(const moveit_msgs::RobotState &state)
 {
+  moveit_msgs::RobotState state_no_attached(state);
+  state_no_attached.attached_collision_objects.clear();
+
   if (parent_)
   {
     if (!kstate_)
     {
       kstate_.reset(new robot_state::RobotState(parent_->getCurrentState()));
       kstate_->setAttachedBodyUpdateCallback(current_state_attached_body_callback_);
     }
-    robot_state::robotStateMsgToRobotState(getTransforms(), state, *kstate_);
+    robot_state::robotStateMsgToRobotState(getTransforms(), state_no_attached, *kstate_);
   }
   else
-    robot_state::robotStateMsgToRobotState(*ftf_, state, *kstate_);
+    robot_state::robotStateMsgToRobotState(*ftf_, state_no_attached, *kstate_);
 
-  // we add object types to the planning scene, if any are specified
```

Comment:
the comment explained the old patch.
The new code calls `processAttachedCollisionObjectMsg` and does not add types here anymore.
The new loop is pretty much self-explaining imho.